vid/src/avid_m/proofs.rs

```diff
@@ -127,6 +133,107 @@ impl MalEncodingProof {
     }
 }
 
+/// A proof of incorrect encoding for a namespace.
+/// It consists of the index of the namespace, the merkle proof of the namespace payload against the namespaced VID commitment,
+/// and the proof of incorrect encoding.
+#[derive(Clone, Debug, Serialize, Deserialize, Eq, PartialEq)]
+pub struct NsAvidMBadEncodingProof {
+    /// The index of the namespace.
+    pub ns_index: usize,
+    /// The commitment of the namespaced VID.
+    pub ns_commit: AvidMCommit,
+    /// The outer merkle proof of the namespace against the namespaced VID commitment.
+    pub ns_mt_proof: MerkleProof,
+    /// The proof of incorrect encoding.
+    pub ns_proof: AvidMBadEncodingProof,
+}
+
+impl NsAvidMScheme {
+    /// Generate a proof of incorrect encoding for a namespace.
+    pub fn proof_of_incorrect_encoding_for_namespace(
+        param: &AvidMParam,
+        ns_index: usize,
+        commit: &NsAvidMCommit,
+        shares: &[NsAvidMShare],
+    ) -> VidResult<NsAvidMBadEncodingProof> {
+        if shares.is_empty() {
+            return Err(VidError::InsufficientShares);
+        }
+        if shares.iter().any(|share| !share.contains_ns(ns_index)) {
+            return Err(VidError::IndexOutOfBound);
+        }
+        let mt = MerkleTree::from_elems(
+            None,
+            shares[0].ns_commits().iter().map(|commit| commit.commit),
+        )?;
+        if mt.commitment() != commit.commit {
+            return Err(VidError::InvalidShare);
+        }
```

Comment:
this is more of `InvalidParam`? indicating that namespace tree's commitment passed in as input argument is wrong, right?

Reply:
you are right. [9d8d35c](https://github.com/EspressoSystems/espresso-network/pull/3292/commits/9d8d35c1ddfe1507fae08b06cfe1f5319ee65661)